src/api/app/jobs/send_event_emails_job.rb

```diff
@@ -32,22 +32,27 @@ def perform
   private
 
   def create_rss_notifications(event)
+    return unless event.eventtype.in?(EVENTS_TO_NOTIFY)
+
     event.subscriptions.each do |subscription|
-      notification_params = {
-        subscriber: subscription.subscriber,
-        event_type: event.eventtype,
-        event_payload: event.payload,
-        subscription_receiver_role: subscription.receiver_role
-      }
-
-      next if subscription.subscriber && subscription.subscriber.away?
-      Notification::RssFeedItem.create(notification_params.merge!(notification_dynamic_params(event)))
+      create_rss_notification(event, subscription)
```

Comment:
Can you move this glue code into the outer loop and out of the rescue? Wrapping your head around singular/pluralized method names is error prone. Like

```ruby
Event::Base.where(mails_sent: false).order(created_at: :asc).limit(1000).each do |event|
  ...
  if event.eventtype.in?(EVENTS_TO_NOTIFY)
    event.subscriptions.each do |subscription|
       create_rss_notification(event, subscription)
    end
  end
end
```

Reply:
Code climate complained when the code was moved outside, without using a method. We changed the method name to `create_notifications` and refactored some more code. Please have a look again.